app/views/dashboard/assessmentDetails.scala.html

```diff
@@ -33,10 +32,36 @@
     </tr>
 }
 
-@table(list: Seq[(String, ApiAssessment)], tableId: String, isDraft: Boolean = false) = {
-    <div id="viewAssessmentRadioGroup_@tableId" class="form-group">
+@effectiveDate(assessment: ApiAssessment) = @{
+    assessment.effectiveDate.map(Formatters.formatDate).getOrElse("")
+}
+
+@fromDate(assessment: ApiAssessment) = @{
+    assessment.currentFromDate.map(Formatters.formatDate).getOrElse("")
+}
+
+@endDate(assessment: ApiAssessment, isDraft: Boolean) = @{
+    if(!isDraft) {
+        (assessment.currentFromDate, assessment.currentToDate) match {
+            case (Some(_), None)     => Some(Messages("assessments.enddate.present"))
+            case (Some(_), Some(to)) => Some(Formatters.formatDate(to))
+            case _                   => None
+        }
+    } else None
+}
+
+@ariaLabel(assessment: ApiAssessment, listType: String, isDraft: Boolean) = @{
+    val listTypeText = listType.takeWhile(_ != 'A').capitalize
+    val fromText = s"from ${fromDate(assessment)}"
+    val endText = endDate(assessment, isDraft).map(ed => s"to $ed").getOrElse("").trim
+
+    s"Select this valuation - $listTypeText $fromText $endText"
+}
+
+@table(list: Seq[(String, ApiAssessment)], listType: String, isDraft: Boolean) = {
+    <div id="viewAssessmentRadioGroup_@listType" class="form-group">
         <div class="table-wrapper-no-pad">
-            <table id="@tableId" class="margin-bottom-10 margin-top-10">
+            <table id="@listType" class="margin-bottom-10 margin-top-10">
```

Comment:
what is `listType` ? is it safe to use this String as an HTML ID attribute?

Reply:
yes - and it hasn't changed - just the name.